var/spack/repos/builtin/packages/scalasca/package.py

```diff
@@ -38,6 +38,9 @@ class Scalasca(Package):
     homepage = "http://www.scalasca.org"
     url = "http://apps.fz-juelich.de/scalasca/releases/scalasca/2.1/dist/scalasca-2.1.tar.gz"
 
+    version('2.3.1', 'a83ced912b9d2330004cb6b9cefa7585',
+            url='http://apps.fz-juelich.de/scalasca/releases/scalasca/2.3/dist/scalasca-2.3.1.tar.gz')
+
     version('2.2.2', '2bafce988b0522d18072f7771e491ab9',
             url='http://apps.fz-juelich.de/scalasca/releases/scalasca/2.2/dist/scalasca-2.2.2.tar.gz')
```

Comment:
I thought someone wrote the opposite - to avoid url_for_version and rely on the fetcher to figure things out. Having said that, I mostly just follow whatever was already there and don't rewrite the entire package.